sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -104,6 +106,14 @@ class UnivocityParser(
     requiredSchema.map(f => makeConverter(f.name, f.dataType, f.nullable, options)).toArray
   }
 
+  private val decimalParser = if (SQLConf.get.legacyDecimalParsing) {
+    (s: String) => new BigDecimal(s.replaceAll(",", ""))
```

Comment:
@MaxGekk, looks we should do the same thing in schema inference path as well.

Reply:
I made some changes but a few tests in `CSVSuite` begun failing because Decimal type is inferred from a timestamps. For example, `2015` from `2015-08-20 15:57:00.0`.

Reply:
I made parsing more strict. Should be fine now.